app/spaces/[spaceid]/trustful/page.tsx

```diff
@@ -0,0 +1,184 @@
+      query GetSpace($id: ID!) {
+        node(id: $id) {
+          ...on Space {
+            avatar
+            banner
+            description
+            name
+            profileId
+            tagline
+            website
+            twitter
+            telegram
+            nostr
+            lens
+            github
+            discord
+            ens
+            admins {
+              id
+            }
+            superAdmin {
+              id
+            }
+            events(first: 10) {
+              edges {
+                node {
+                  createdAt
+                  description
+              endTime
+              timezone
+              status
+              tagline
+              imageUrl
+              externalUrl
+              gated
+              id
+              meetingUrl
+              profileId
+              spaceId
+              startTime
+              title
+              space {
+                avatar
+                name
+              }
+                }
+              }
+            }
+          }
+        }
+      }
+      `;
+
+    const response: any = await composeClient.executeQuery(GET_SPACE_QUERY, {
+      id: spaceId,
+    });
+    const spaceData: Space = response.data.node as Space;
+    setSpace(spaceData);
+    const eventData: SpaceEventData = response.data.node
+      .events as SpaceEventData;
+    const fetchedEvents: Event[] = eventData.edges.map((edge) => edge.node);
+    setEvents(fetchedEvents);
+    return spaceData;
+  };
+  useEffect(() => {
+    const fetchData = async () => {
+      const space = await getSpaceByID();
+      document.title = space?.name + ' - ' + 'Zuzalu City';
+      const admins =
+        space?.admins?.map((admin) => admin.id.toLowerCase()) || [];
+      const superAdmins =
+        space?.superAdmin?.map((superAdmin) => superAdmin.id.toLowerCase()) ||
+        [];
+      const userDID = ceramic?.did?.parent.toString().toLowerCase() || '';
+      if (admins.includes(userDID) || superAdmins.includes(userDID)) {
+        setIsAdmin(true);
+      }
+    };
+
+    fetchData()
+      .catch((error) => {
+        console.error('An error occurred:', error);
+      })
+      .finally(() => {
+        setIsEventsLoading(false);
+      });
+  }, []);
+
+  const address = useMemo(() => {
+    if (profile) {
+      const id = profile.author?.id.split(':');
+      return id?.[id?.length - 1];
+    }
+  }, [profile]);
+
+  useEffect(() => {
+    const fetchUserRole = async () => {
+      if (address) {
+        await getUserRole(address as Address)
+          .then((data) => {
+            console.log('User role:', data);
+            if (data && data?.role) {
+              setUserRole({
+                address: address as Address,
+                role: data.role,
+              });
+            }
+          })
+          .catch((error) => console.error('Failed to fetch events:', error));
+      }
+    };
+    fetchUserRole();
+  }, []);
```

Comment:
Seems like this needs to refetch on address change

Reply:
Damm bro, I was thinking you will not catch that ! I added in another branch 😆  But yeah, let me add here as well :)

Reply:
You smart boy! 😂